pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -125,58 +131,130 @@ func (ku *KubeUtil) searchPodForContainerID(podlist []*Pod, containerID string)
 	return nil, fmt.Errorf("container %s not found in podlist", containerID)
 }
 
-// Try and find the hostname to query the kubelet
-// TODO: Add TLS verification
-func locateKubelet() (string, error) {
-	host := config.Datadog.GetString("kubernetes_kubelet_host")
-	if host == "" {
-		var err error
-		host, err = docker.HostnameProvider("")
-		if err != nil {
-			return "", fmt.Errorf("unable to get hostname from docker, please set the kubernetes_kubelet_host option: %s", err)
-		}
+// setupKubeletApiClient will try to setup the http(s) client to query the kubelet
+// with the following settings, in order:
+//  - Load Certificate Authority if needed
+//  - HTTPS w/ configured certificates
+//  - HTTPS w/ configured token
+//  - HTTPS w/ service account token
+//  - HTTP
+func (ku *KubeUtil) setupKubeletApiClient() error {
+	tlsConfig, err := getTLSConfig()
+	if err != nil {
+		return err
 	}
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	ku.kubeletApiClient.Transport = transport
+
+	switch {
+	case isConfiguredCertificates():
+		tlsConfig.Certificates, err = kubernetes.GetCertificates(
+			config.Datadog.GetString("kubelet_client_crt"),
+			config.Datadog.GetString("kubelet_client_key"),
+		)
+		return err
+
+	case isConfiguredTokenPath():
+		return ku.setBearerToken(config.Datadog.GetString("kubelet_auth_token_path"))
 
-	port := config.Datadog.GetInt("kubernetes_http_kubelet_port")
-	url := fmt.Sprintf("http://%s:%d", host, port)
-	healthzURL := fmt.Sprintf("%s%s", url, KubeletHealthPath)
-	if _, err := PerformKubeletQuery(healthzURL); err == nil {
-		return url, nil
+	case kubernetes.IsServiceAccountToken():
```

Comment:
The default for **kubelet_auth_token_path** is `""`. So if an advertised user set a custom token path and he made a mistake about it, do we still want to fallback to something not intend in the first place ?

Reply:
if `kubelet_auth_token_path` is set to a non-default value, we want to fail hard but explicitly, instead of fallback-ing to the serviceaccount. Like like the logic actually does that, right?

Reply:
Actually it does exactly that:
> if kubelet_auth_token_path is set to a non-default value, we want to fail hard but explicitly, instead of fallback-ing to the serviceaccount.

This is properly tested in the unittests.
I'm adding also a similar high level testing scenario in the integration tests.